File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/videos/GameVideosFragment.java
package com.exact.twitch.ui.videos;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.View;

import com.exact.twitch.R;

public class GameVideosFragment extends BaseVideosFragment implements VideosSortDialog.OnSortOptionSelected {

    private String game;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (isFragmentVisible()) {
            if (getViewModel().getCurrentOrderText().getValue() == null) {
                getViewModel().getCurrentOrderText().postValue(getString(R.string.view_count));
            }
            if (getViewModel().getSelectedOrderItem() == null) {
                getViewModel().setSelectedOrderItem(R.id.fragment_video_sort_dialog_views);
            }
            if (getViewModel().getCurrentPeriodText().getValue() == null) {
                getViewModel().getCurrentPeriodText().postValue(getString(R.string.this_week));
            }
            if (getViewModel().getSelectedPeriodItem() == null) {
                getViewModel().setSelectedPeriodItem(R.id.fragment_video_sort_dialog_week);
            }
            if (getViewModel().getSort() == null) {
                getViewModel().setSort("views");
            }
            getViewModel().getCurrentOrderText().observe(this, text -> currentSortOption.setText(getString(R.string.order_and_period, text, getViewModel().getCurrentPeriodText().getValue())));
            getViewModel().getCurrentPeriodText().observe(this, text -> currentSortOption.setText(getString(R.string.order_and_period, getViewModel().getCurrentOrderText().getValue(), text)));
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (getArguments() != null) {
            game = getArguments().getString("game");
        }
        sortByContainer.setOnClickListener(v -> VideosSortDialog.newInstance(getViewModel().getSelectedOrderItem(), getViewModel().getSelectedPeriodItem())                .show(getChildFragmentManager(), null));
    }

    @Override
    public void onSort(int orderItemId, CharSequence orderText, int periodItemId, CharSequence periodText) {
        boolean shouldReload = false;
        if (getViewModel().getSelectedOrderItem() != orderItemId) {
            shouldReload = true;
            getViewModel().setSort(orderItemId == R.id.fragment_video_sort_dialog_time ? "time" : "views");
            getViewModel().getCurrentOrderText().postValue(orderText);
            getViewModel().setSelectedOrderItem(orderItemId);
        }
        if (getViewModel().getSelectedPeriodItem() != periodItemId) {
            shouldReload = true;
            String period = null;
            switch (periodItemId) {
                case R.id.fragment_video_sort_dialog_week:
                    period = "week";
                    break;
                case R.id.fragment_video_sort_dialog_month:
                    period = "month";
                    break;
                case R.id.fragment_video_sort_dialog_all:
                    period = "all";
                    break;
            }
            getViewModel().setSelectedPeriodItem(periodItemId);
            getViewModel().setPeriod(period);
            getViewModel().getCurrentPeriodText().postValue(periodText);
        }
        if (shouldReload) {
            loadData(true);
        }
    }

    @Override
    public void loadData(boolean override) {
        getViewModel().loadVideos(game, getViewModel().getPeriod(), null, null, getViewModel().getSort(), override);
    }

    @Override
    protected void initData() {
        loadData(false);
    }
}

File: app/src/main/java/com/github/exact7/xtra/ui/videos/GameVideosFragment.kt
package com.github.exact7.xtra.ui.videos

import android.os.Bundle
import android.view.View
import com.github.exact7.xtra.R
import com.github.exact7.xtra.model.game.Game
import kotlinx.android.synthetic.main.fragment_videos.*

class GameVideosFragment : BaseVideosFragment(), VideosSortDialog.OnFilterApplied {

    private lateinit var game: Game

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        game = arguments?.getParcelable("game")!!
        sortBar.setOnClickListener { VideosSortDialog.newInstance(viewModel.sort, viewModel.period!!).show(childFragmentManager, null) }
    }

    override fun initializeViewModel() {
        viewModel.sort = Sort.VIEWS
        viewModel.period = Period.WEEK
        setSortText(getString(R.string.view_count), getString(R.string.this_week))
    }

//    override fun loadData(override: Boolean) {
//        viewModel.loadVideos(game = game.info.name, reload = override)
//    }

    override fun onApply(sort: Sort, sortText: CharSequence, period: Period, periodText: CharSequence) {
        var shouldReload = false
        if (viewModel.sort != sort) {
            shouldReload = true
            viewModel.sort = sort
        }
        if (viewModel.period != period) {
            shouldReload = true
            viewModel.period = period
        }
        if (shouldReload) {
            setSortText(sortText, periodText)
//            viewModel.loadedInitial.value = null
            adapter.submitList(null)
//            loadData(true)
        }
    }

    private fun setSortText(sort: CharSequence, period: CharSequence) {
        viewModel.sortText.postValue(getString(R.string.sort_and_period, sort, period))
    }
}
